capa/features/extractors/vmray/__init__.py

```diff
@@ -128,41 +147,42 @@ def _compute_sections(self):
             for elffile_section in self.sample_file_static_data.elf.sections:
                 self.sections[elffile_section.header.sh_addr] = elffile_section.header.sh_name
 
-    def _compute_process_ids(self):
+    def _compute_monitor_processes(self):
         for process in self.sv2.processes.values():
-            # we expect VMRay's monitor IDs to be unique, but OS PIDs may be reused
-            assert process.monitor_id not in self.os_pid_by_monitor_id.keys()
-            self.os_pid_by_monitor_id[process.monitor_id] = process.os_pid
+            # we expect monitor IDs to be unique
+            assert process.monitor_id not in self.monitor_processes.keys()
 
-        # not all processes may get an ID, get missing data from flog.xml, see #2394
+            ppid: int = (
+                self.sv2.processes[process.ref_parent_process.path[1]].os_pid if process.ref_parent_process else 0
+            )
+            self.monitor_processes[process.monitor_id] = VMRayMonitorProcess(
+                process.os_pid, ppid, process.monitor_id, process.image_name
+            )
+
+        # not all processes are recorded in SummaryV2.json, get missing data from flog.xml, see #2394
         for monitor_process in self.flog.analysis.monitor_processes:
-            if monitor_process.process_id in self.os_pid_by_monitor_id:
-                assert self.os_pid_by_monitor_id[monitor_process.process_id] == monitor_process.os_pid
-            else:
-                self.os_pid_by_monitor_id[monitor_process.process_id] = monitor_process.os_pid
-
-    def _compute_process_threads(self):
-        # logs/flog.xml appears to be the only file that contains thread-related data
-        # so we use it here to map processes to threads
-        for function_call in self.flog.analysis.function_calls:
-            pid: int = self.get_process_os_pid(function_call.process_id)  # flog.xml uses process monitor ID, not OS PID
-            tid: int = function_call.thread_id
+            if monitor_process.process_id not in self.monitor_processes.keys():
```

Comment:
should we perform some sanity checks on the existing data, i.e., that the IDs line up?

Reply:
https://github.com/mandiant/capa/pull/2396/commits/df878c78576ae66444bb0b49257d1f27ba4d5d91